### ISSUE TITLE
[BUG] wrong results in using-feature-columns.ipynb nb after Preprocessing with NVT 

### ISSUE THREAD

**Issue by rnyak** [NVIDIA/NVTabular#567]
**Describe the bug**

The using-feature-columns.ipynb nb is giving wrong results after doing the first preprocessing step which is as below:

```
cat_names = ColumnGroup(
    [
        "Type",
        "Breed1",
        "Color1",
        "Color2",
        "MaturitySize",
        "Gender",
        "FurLength",
        "Vaccinated",
        "Sterilized",
        "Health",
    ]
)
cont_names = ColumnGroup(["Age", "Fee", "PhotoAmt"])
label = ColumnGroup(["AdoptionSpeed"])

label_feature = label >> (lambda col: cp.where(col == 4, 0, 1))
workflow = nvt.Workflow(cat_names + cont_names + label_feature)

for split in ["train", "valid", "test"]:
    output_path = os.path.join(data_dir, "processed", split)
    if not os.path.exists(output_path):
        os.makedirs(output_path)

    dataset = nvt.Dataset(os.path.join(data_dir, split + ".parquet"))
    if split == "train":
        shuffle = nvt.io.Shuffle.PER_WORKER
        workflow.fit(dataset)
        workflow.transform(dataset).to_parquet(
            output_path=output_path, shuffle=shuffle, out_files_per_proc=1
        )
    else:
        shuffle = False
        workflow.transform(dataset).to_parquet(
            output_path=output_path, shuffle=shuffle, out_files_per_proc=1
        )
```
The issues:

1. the original train, valid and test sets do not have null values for the feautes that we use in training, but after doing preprocessing with NVT, we get some `<NA>` for each column.
2.  Although we set  `out_files_per_proc=1` both for train and valid, we get `2` output parquet files in the output path.
3. The final gdfs total num_rows is grater than the original file's num_rows. 

**Steps/Code to reproduce bug**
To reproduce the issues above, run the using-feature-columns.ipynb nb and check the preprocessed 

**Environment details (please complete the following information):**
 - Environment location: [Bare-metal, Docker, Cloud(specify cloud provider)]: Docker container
 - Method of NVTabular install: [conda, Docker, or from source] Docker
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used
```
docker run --gpus=3 -it -p 9999:8888 -p 9797:8787 -p 9796:8786 --ipc=host --cap-add SYS_PTRACE gitlab-master.nvidia.com:5005/rapidsdl/docker/rapidsdl_joc:latest
 ```
I also ran the notebook in a conda env using rapids 0.16 and rapids 0.17. I have had the same issues.

**Comment by rnyak** [comment]
@rjzamora would you mind looking into the issues I reported above? Did we change anything regarding to writing to parquet the output files?  Do you have any idea about the problem? Thanks for your time.

**Comment by rjzamora** [comment]
@rnyak - I wasn't able to reproduce the file-count problem (I always get 1 file with `out_files_per_proc=1`), but I did notice that index misalignment was causing the other issues during column concatenation.  I added a fix for this to #566 -  Could you check if the problems are resolved in that branch?

**Comment by rnyak** [comment]
@benfred closing. It is now working after the fix in https://github.com/NVIDIA/NVTabular/pull/566.
